x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/RealmConfig.java

```diff
@@ -24,11 +24,24 @@
     private final ThreadContext threadContext;
 
     public RealmConfig(RealmIdentifier identifier, Settings settings, Environment env, ThreadContext threadContext) {
+        this(identifier, settings, env, threadContext, null);
+    }
+
+    public RealmConfig(RealmIdentifier identifier, Settings settings, Environment env, ThreadContext threadContext, Integer order) {
         this.identifier = identifier;
         this.settings = settings;
         this.env = env;
         this.enabled = getSetting(RealmSettings.ENABLED_SETTING);
-        this.order = getSetting(RealmSettings.ORDER_SETTING);
+        if (order != null) {
+            this.order = order;
+        } else if (order == null && hasSetting(RealmSettings.ORDER_SETTING.apply(type())) == false) {
+            throw new IllegalArgumentException("'order' is a mandatory parameter for realm config. " +
+                "Found invalid realm config: '" + identifier.name + "'\n" +
+                "Please see the breaking changes documentation."
+            );
+        } else {
+            this.order = getSetting(RealmSettings.ORDER_SETTING);
+        }
```

Comment:
I would prefer the `order` be a primitive, and we resolve the order from the settings in the original constructor.

Behaviours around _if this parameter is null, do this other thing_ can end up getting quite confusing in a large code base.

Reply:
I agree `null` is a hack. But there is no easy way to tell whether a primitive value is indeed missing.

Reply:
Changed to `int`.